### PR TITLE
Added Clickable Links to narrow for user!

### LINKED ISSUE #39: Narrow to @ messages
Is there a way to narrow to @-messages in the app?  Perusing the app, I haven't been able to find one.

### PR DESCRIPTION
**Pull request by kunall17**
Fixes #39 

device-2016-07-21-020805

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@kunall17, it looks like you've already signed the Dropbox CLA. Thanks!

**Comment by niftynei** [comment]
@kunall17 when I tap on a link, I get the following crash:

```
07-28 15:32:07.130 17641 17641 E AndroidRuntime: FATAL EXCEPTION: main
07-28 15:32:07.130 17641 17641 E AndroidRuntime: Process: com.zulip.android.dev, PID: 17641
07-28 15:32:07.130 17641 17641 E AndroidRuntime: java.lang.ClassCastException: android.support.v7.widget.TintContextWrapper cannot be cast to com.zulip.android.activities.ZulipActivity
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at com.zulip.android.util.ProfileSpan.onClick(ProfileSpan.java:43)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.text.method.LinkMovementMethod.onTouchEvent(LinkMovementMethod.java:217)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.widget.TextView.onTouchEvent(TextView.java:8325)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.View.dispatchTouchEvent(View.java:9300)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2553)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2553)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2553)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2553)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2553)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2553)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2553)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2553)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2553)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2553)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2553)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2553)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2553)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2254)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at com.android.internal.policy.PhoneWindow$DecorView.superDispatchTouchEvent(PhoneWindow.java:2403)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1737)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.app.Activity.dispatchTouchEvent(Activity.java:2771)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.support.v7.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:63)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.support.v7.view.WindowCallbackWrapper.dispatchTouchEvent(WindowCallbackWrapper.java:63)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at com.android.internal.policy.PhoneWindow$DecorView.dispatchTouchEvent(PhoneWindow.java:2364)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.View.dispatchPointerEvent(View.java:9520)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:4238)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4104)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3650)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3703)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3669)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:3795)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3677)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:3852)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3650)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3703)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3669)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3677)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3650)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:5932)
07-28 15:32:07.130 17641 17641 E AndroidRuntime: 	at android.vi
```

**Comment by niftynei** [comment]
This looks good to me Kunal, other than the crash.

File: app/src/main/java/com/zulip/android/util/CustomHtmlToSpannedConverter.java [hunk 1]
```diff
@@ -67,17 +71,18 @@
     private Html.TagHandler mTagHandler;
     private Html.ImageGetter mEmojiGetter;
     private String mBaseUri;
-
+    private static int userMentionColor;
```
Comment (niftynei) [inline app/src/main/java/com/zulip/android/util/CustomHtmlToSpannedConverter.java]: does this need to be static?

**Comment by kunall17** [comment]
@niftynei updated..:)

**Comment by kunall17** [comment]
@niftynei updated this branch as well!

**Comment by niftynei** [comment]
Two issues with this:

1) I click on @kunall17 , only a blank page shows. It works correctly for other users with no messages and with messages.
2) Pushing the back button doesn't take me back to the message that I clicked from, but resets it all the way back to the default `ZulipActivity` view.  I'd expect to return to the place that I clicked into the user messages from.  This is a blocker to getting this merged.

**Comment by kunall17** [comment]
@nitfynei f8ca6c3 commit fixes first issue (when you clicked the link the toolbar was hidden, therefore a blank page would appear) now this commit makes the toolbar appear whenever narrowing is done! 

second issue will be fixed in another PR which will take back to previous location after narrowing!

**Comment by niftynei** [comment]
opened issue #146 for the losing your place on back issue.

**Comment by niftynei** [comment]
@kunall17 I still get a blank screen. Tested by tapping on your user name :)

**Comment by niftynei** [comment]
The functionality that was asked for in #39 was to add the `@` button, like on the web app:

image

**Comment by kunall17** [comment]
@niftynei I thought in the issue you were talking about the clickable links! 
Anyways this was also needed, fixed the issue!

**Comment by kunall17** [comment]
Updated this branch! 
Any blocker for this branch?

**Comment by niftynei** [comment]
Merged!!!!! great piece of work @kunall17 🔨 🔨

**Pull request closed without merging**